File: iosApp/CocktailCraft/Utils/PlatformErrorHandler.swift
import Foundation
import shared

@MainActor
class PlatformErrorHandler {
    static let shared = PlatformErrorHandler()
    
    private init() {}
    
    func getErrorFromPlatformException(
        exception: NSError,
        defaultMessage: String,
        recoveryAction: ErrorHandler.RecoveryAction?
    ) -> ErrorHandler.UserFriendlyError {
        // Map iOS-specific errors to UserFriendlyError
        switch exception.code {
        case NSURLErrorNotConnectedToInternet,
             NSURLErrorNetworkConnectionLost:
            return ErrorHandler.UserFriendlyError(
                title: "Network Error",
                message: "Unable to connect to the server. Please check your internet connection.",
                category: ErrorHandler.ErrorCategory.network,
                recoveryAction: recoveryAction,
                originalException: nil,
                errorCode: .network
            )
            
        case NSURLErrorTimedOut:
            return ErrorHandler.UserFriendlyError(
                title: "Connection Timeout",
                message: "The connection timed out. Please try again.",
                category: ErrorHandler.ErrorCategory.network,
                recoveryAction: recoveryAction,
                originalException: nil,
                errorCode: .timeout
            )
            
        case NSURLErrorCannotFindHost,
             NSURLErrorCannotConnectToHost:
            return ErrorHandler.UserFriendlyError(
                title: "Server Error",
                message: "Unable to reach the server. Please try again later.",
                category: ErrorHandler.ErrorCategory.server,
                recoveryAction: recoveryAction,
                originalException: nil,
                errorCode: .serverError
            )
            
        default:
            // Fallback to shared error handler
            return ErrorHandler.shared.createUserFriendlyError(
                title: "Error",
                message: exception.localizedDescription.isEmpty ? defaultMessage : exception.localizedDescription,
                category: ErrorHandler.ErrorCategory.unknown,
                recoveryAction: recoveryAction,
                originalException: nil,
                errorCode: .unknown
            )
        }
    }
}

File: shared/src/androidMain/kotlin/com/cocktailcraft/util/PlatformErrorHandler.kt
package com.cocktailcraft.util

import java.net.ConnectException
import java.net.SocketTimeoutException
import java.net.UnknownHostException
import java.util.concurrent.TimeoutException

/**
 * Android-specific error handling extensions
 */
object PlatformErrorHandler {
    
    /**
     * Convert Android-specific network exceptions to user-friendly errors
     */
    fun getErrorFromPlatformException(
        exception: Throwable,
        defaultMessage: String = "Something went wrong. Please try again.",
        recoveryAction: ErrorHandler.RecoveryAction? = null
    ): ErrorHandler.UserFriendlyError {
        return when (exception) {
            // Network errors
            is UnknownHostException,
            is ConnectException -> ErrorHandler.UserFriendlyError(
                title = "Network Error",
                message = "Unable to connect to the server. Please check your internet connection and try again.",
                category = ErrorHandler.ErrorCategory.NETWORK,
                recoveryAction = recoveryAction,
                originalException = exception,
                errorCode = com.cocktailcraft.domain.util.ErrorCode.NETWORK
            )

            // Timeout errors
            is SocketTimeoutException,
            is TimeoutException -> ErrorHandler.UserFriendlyError(
                title = "Connection Timeout",
                message = "The connection timed out. This might be due to slow internet or server issues. Please try again.",
                category = ErrorHandler.ErrorCategory.NETWORK,
                recoveryAction = recoveryAction,
                originalException = exception,
                errorCode = com.cocktailcraft.domain.util.ErrorCode.TIMEOUT
            )

            // Fallback to common error handler
            else -> ErrorHandler.getErrorFromException(exception, defaultMessage, recoveryAction)
        }
    }
}